### PR TITLE
Add PSR compatible BigQueryLogger

### PR DESCRIPTION
**Pull request by akleineb**
This PR adds an PSR compatible BigQueryLogger which creates a partitioned table in BigQuery where you can hold a massive amount of logs.

### CODE REVIEW

File: Classes/Service/BigQueryService.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    /**
+     * @var BigQueryClient
+     */
+    protected $bigQueryClient;
+
+    public function getClient(): BigQueryClient
+    {
+        if ($this->settings['keyFilePath'] === null || $this->settings['dataset'] === null || $this->settings['table'] === null) {
+            throw new Exception('BigQueryLogger is not configured.');
```
Comment (johannessteu) [inline Classes/Service/BigQueryService.php]:
```suggestion
            throw new Exception('BigQueryLogger is not correctly configured. Make sure to set keyFilePath, dataset and table.');
```

File: Classes/Service/BigQueryService.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+     * @var mixed[]
+     */
+    protected $settings;
+
+    /**
+     * @var BigQueryClient
+     */
+    protected $bigQueryClient;
+
+    public function getClient(): BigQueryClient
```
Comment (johannessteu) [inline Classes/Service/BigQueryService.php]:
```suggestion
    protected function getClient(): BigQueryClient
```

File: Classes/Service/BigQueryService.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        if ($this->settings['keyFilePath'] === null || $this->settings['dataset'] === null || $this->settings['table'] === null) {
```
Comment (johannessteu) [inline Classes/Service/BigQueryService.php]: Checks should check for empty values as well

File: Classes/Service/BigQueryService.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    protected function getDataset(): Dataset
+    {
+        return $this->getClient()->dataset($this->settings['dataset']);
+    }
+
+    /**
+     * @return mixed[]
+     */
+    protected function getTableSchema(): array
```
Comment (johannessteu) [inline Classes/Service/BigQueryService.php]: Move this to Settings.yaml to make it easy configurable?

File: README.md [hunk 5]
```diff
@@ -33,3 +33,36 @@ t3n:
...
+    log:
+      psr3:
+        'Neos\Flow\Log\PsrLoggerFactory':
+          bigQueryLogger:
+            default:
+              class: 't3n\FlowLog\Backend\BigQueryLogger'
+              options:
+                loggerName: 'applicationXyImportLogger'
+```
+
```
Comment (johannessteu) [inline README.md:69]: Missing documentation about how to actually use a logger

**Review by johannessteu — CHANGES_REQUESTED** [review]

File: Classes/Service/BigQueryService.php (hunk 3 above)
Comment (akleineb) [reply Classes/Service/BigQueryService.php]: Good point - fixed!

File: Classes/Service/BigQueryService.php (hunk 4 above)
Comment (akleineb) [reply Classes/Service/BigQueryService.php]: I have moved the table schema. Now everybody is able to extend the logging behaviour.

File: README.md (hunk 5 above)
Comment (akleineb) [reply README.md:69]: I have extended the readme. Have a look 👀😄

File: README.md [hunk 6]
```diff
@@ -24,7 +24,92 @@ Neos:
...
+          1:
+            value: bigQueryLogger
+
+```
+
+Now you are able to use it:
+
+```php
+/**
+ * @var Neos\Flow\Log\Psr\Logger
```
Comment (johannessteu) [inline README.md]: Should be LoggerInterface, right?

**Review by johannessteu — CHANGES_REQUESTED** [review]

**Review by johannessteu — APPROVED** [review]

**Pull request merged**